Embulk 0.10.35 with JRuby 9.3.3.0 raise the NoMethodError with "undefined method `runtime'"

Issue Type: Bug Report
-----------------------

* Write the following environmental information.
  * OS version: macOS 12.3.1
  * Java version: 1.8.0_291
  * Embulk version: 0.10.35
  * Your Embulk configuration (YAML)
  * Plugin versions
    * embulk-input-postgresql: 0.12.3
* Write all what you did, e.g. your commands executed
* Write what you expected and observed actually


Embulk 0.10.35 with JRuby 9.3.3.0 raise the following exception when a java plugin load.

```
org.jruby.exceptions.NoMethodError: (NoMethodError) undefined method `runtime' for JRuby:Module
```

It relates to [this issue](https://github.com/msgpack/msgpack-ruby/issues/226).

```
cat ~/.embulk-dev/embulk.properties
jruby=file:///Users/user/.embulk/jars/jruby-complete-9.3.3.0.jar
```

```
java -jar ~/.embulk/bin/embulk-0.10.x -X embulk_home=$HOME/.embulk-dev run conf.yml
2022-04-15 21:17:07.288 +0900 [INFO] (main): embulk_home is set from command-line: /Users/user/.embulk-dev
2022-04-15 21:17:07.291 +0900 [INFO] (main): m2_repo is set as a sub directory of embulk_home: /Users/user/.embulk-dev/lib/m2/repository
2022-04-15 21:17:07.292 +0900 [INFO] (main): gem_home is set as a sub directory of embulk_home: /Users/user/.embulk-dev/lib/gems
2022-04-15 21:17:07.292 +0900 [INFO] (main): gem_path is set empty.
2022-04-15 21:17:07.630 +0900 [INFO] (main): Started Embulk v0.10.35
2022-04-15 21:17:11.357 +0900 [INFO] (0001:transaction): Gem's home and path are set by system configs "gem_home": "/Users/user/.embulk-dev/lib/gems", "gem_path": ""
org.embulk.config.ConfigException: InputPlugin 'postgresql' is not found.
Maven repository specified is not found at "/Users/user/.embulk-dev/lib/m2/repository" ("/Users/user/.embulk-dev/lib/m2/repository").
org.jruby.exceptions.NoMethodError: (NoMethodError) undefined method `runtime' for JRuby:Module
	at org.embulk.plugin.PluginManager.buildPluginNotFoundException(PluginManager.java:155)
	at org.embulk.plugin.PluginManager.newPlugin(PluginManager.java:99)
	at org.embulk.spi.ExecSessionInternal.newPlugin(ExecSessionInternal.java:358)
	at org.embulk.spi.ExecInternal.newPlugin(ExecInternal.java:50)
	at org.embulk.exec.BulkLoader$ProcessPluginSet.<init>(BulkLoader.java:440)
	at org.embulk.exec.BulkLoader.doRun(BulkLoader.java:515)
	at org.embulk.exec.BulkLoader.access$000(BulkLoader.java:35)
	at org.embulk.exec.BulkLoader$1.run(BulkLoader.java:352)
	at org.embulk.exec.BulkLoader$1.run(BulkLoader.java:349)
	at org.embulk.spi.ExecInternal.doWith(ExecInternal.java:26)
	at org.embulk.exec.BulkLoader.run(BulkLoader.java:349)
	at org.embulk.EmbulkEmbed.run(EmbulkEmbed.java:278)
	at org.embulk.EmbulkRunner.runInternal(EmbulkRunner.java:288)
	at org.embulk.EmbulkRunner.run(EmbulkRunner.java:153)
	at org.embulk.cli.EmbulkRun.runInternal(EmbulkRun.java:118)
	at org.embulk.cli.EmbulkRun.run(EmbulkRun.java:24)
	at org.embulk.cli.Main.main(Main.java:55)
	Suppressed: org.embulk.plugin.PluginSourceNotMatchException
		at org.embulk.plugin.BuiltinPluginSource.newPlugin(BuiltinPluginSource.java:244)
		at org.embulk.plugin.PluginManager.newPlugin(PluginManager.java:76)
		... 15 more
	Suppressed: org.embulk.plugin.PluginSourceNotMatchException: org.embulk.deps.maven.MavenRepositoryNotFoundException: Maven repository specified is not found at "/Users/user/.embulk-dev/lib/m2/repository" ("/Users/user/.embulk-dev/lib/m2/repository").
		at org.embulk.plugin.maven.MavenPluginRegistry.findPluginPaths(MavenPluginRegistry.java:102)
		at org.embulk.plugin.maven.MavenPluginRegistry.lookup(MavenPluginRegistry.java:71)
		at org.embulk.plugin.maven.MavenPluginSource.newPlugin(MavenPluginSource.java:33)
		at org.embulk.plugin.PluginManager.newPlugin(PluginManager.java:82)
		... 15 more
	Caused by: org.embulk.deps.maven.MavenRepositoryNotFoundException: Maven repository specified is not found at "/Users/user/.embulk-dev/lib/m2/repository" ("/Users/user/.embulk-dev/lib/m2/repository").
		at org.embulk.deps.maven.MavenArtifactFinderImpl.<init>(MavenArtifactFinderImpl.java:44)
		at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
		at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
		at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
		at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
		at org.embulk.deps.maven.MavenArtifactFinder.create(MavenArtifactFinder.java:13)
		at org.embulk.plugin.maven.MavenPluginRegistry.findPluginPaths(MavenPluginRegistry.java:100)
		... 18 more
	Caused by: java.nio.file.NoSuchFileException: /Users/user/.embulk-dev/lib/m2/repository
		... 25 more
	Suppressed: org.embulk.plugin.PluginSourceNotMatchException
		at org.embulk.plugin.SelfContainedPluginRegistry.search(SelfContainedPluginRegistry.java:77)
		at org.embulk.plugin.SelfContainedPluginRegistry.lookup(SelfContainedPluginRegistry.java:64)
		at org.embulk.plugin.SelfContainedPluginSource.newPlugin(SelfContainedPluginSource.java:44)
		at org.embulk.plugin.PluginManager.newPlugin(PluginManager.java:88)
		... 15 more
	Suppressed: org.embulk.plugin.PluginSourceNotMatchException: Failed to initialize embulk-input-postgresql.
		at org.embulk.jruby.JRubyPluginSource.newPlugin(JRubyPluginSource.java:89)
		at org.embulk.plugin.PluginManager.newPlugin(PluginManager.java:94)
		... 15 more
	Caused by: org.jruby.embed.InvokeFailedException: org.jruby.exceptions.NoMethodError: (NoMethodError) undefined method `runtime' for JRuby:Module
		at org.jruby.embed.internal.EmbedRubyObjectAdapterImpl.doInvokeMethod(EmbedRubyObjectAdapterImpl.java:253)
		at org.jruby.embed.internal.EmbedRubyObjectAdapterImpl.callMethod(EmbedRubyObjectAdapterImpl.java:158)
		at org.jruby.embed.ScriptingContainer.callMethod(ScriptingContainer.java:1448)
		at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
		at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.lang.reflect.Method.invoke(Method.java:498)
		at org.embulk.jruby.ScriptingContainerDelegateImpl.callMethod(ScriptingContainerDelegateImpl.java:634)
		at org.embulk.jruby.LazyScriptingContainerDelegate.callMethod(LazyScriptingContainerDelegate.java:227)
		at org.embulk.jruby.JRubyPluginSource.newPlugin(JRubyPluginSource.java:87)
		... 16 more
	Caused by: org.jruby.exceptions.NoMethodError: (NoMethodError) undefined method `runtime' for JRuby:Module
		at RUBY.classloader(/Users/user/.embulk-dev/lib/gems/gems/embulk-0.10.35-java/lib/embulk/java_plugin.rb:18)
		at RUBY.register_input(/Users/user/.embulk-dev/lib/gems/gems/embulk-0.10.35-java/lib/embulk/java_plugin.rb:22)
		at RUBY.<main>(/Users/user/.embulk-dev/lib/gems/gems/embulk-input-postgresql-0.12.3-java/lib/embulk/input/postgresql.rb:1)
		at org.jruby.RubyKernel.require(org/jruby/RubyKernel.java:1017)
		at uri_3a_classloader_3a_.META_minus_INF.jruby_dot_home.lib.ruby.stdlib.rubygems.core_ext.kernel_require.require(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:160)
		at RUBY.require_and_show(/Users/user/.embulk-dev/lib/gems/gems/embulk-0.10.35-java/lib/embulk/plugin_registry.rb:81)
		at RUBY.search(/Users/user/.embulk-dev/lib/gems/gems/embulk-0.10.35-java/lib/embulk/plugin_registry.rb:39)
		at RUBY.lookup(/Users/user/.embulk-dev/lib/gems/gems/embulk-0.10.35-java/lib/embulk/plugin_registry.rb:26)
		at RUBY.lookup(/Users/user/.embulk-dev/lib/gems/gems/embulk-0.10.35-java/lib/embulk/plugin.rb:197)
		at RUBY.new_java_input(/Users/user/.embulk-dev/lib/gems/gems/embulk-0.10.35-java/lib/embulk/plugin.rb:155)
		at RUBY.new_java_input(uri:classloader:/META-INF/jruby.home/lib/ruby/stdlib/forwardable.rb:230)

Error: InputPlugin 'postgresql' is not found.
Maven repository specified is not found at "/Users/user/.embulk-dev/lib/m2/repository" ("/Users/user/.embulk-dev/lib/m2/repository").
org.jruby.exceptions.NoMethodError: (NoMethodError) undefined method `runtime' for JRuby:Module
```

```yaml
in:
  type: postgresql
  host: localhost
  user: user
  password: password
  database: embulk_test
  table: sample
out:
  type: stdout
```

```
embulk_test=# select * from sample;
 id | col_text | col_double |        col_time
----+----------+------------+------------------------
  1 | text     |      1.234 | 2017-04-12 00:00:00+09
(1 row)
```

--- a/embulk-ruby/lib/embulk/java/imports.rb
+++ b/embulk-ruby/lib/embulk/java/imports.rb
@@ -1,4 +1,19 @@
+# Required to use "Java features" from the Ruby side.
 require 'java'
+
+# Required to access JRuby-specific internal features, such as `JRuby.runtime`.
+require 'jruby' if Gem::Version.new(JRUBY_VERSION) >= Gem::Version.new('9.3')
+
+# Both of 'java' and 'jruby' should be loaded in Embulk's use-case in nature.
+#
+# Embulk had loaded only 'java' explicitly until Embulk v0.10.35.
+# It was because loading 'java' automatically loaded also 'jruby' until JRuby 9.2.
+# But, it has changed since JRuby 9.3. JRuby 9.3+ needs loading 'jruby' explicitly.
+#
+# See also: https://github.com/jruby/jruby/issues/7221#issuecomment-1133646241
+#
+# Here, Embulk limits loading 'jruby' only in JRuby 9.3+ to ensure compatibility.
+# We may remove the limitation in the future.
 
 #
 # this file is loaded by embulk/java/bootstrap.rb

